charts/seq/templates/deployment.yaml

```diff
@@ -39,9 +39,18 @@ spec:
             - name: "SEQ_FIRSTRUN_ADMINUSERNAME"
               value: "{{ .Values.firstRunAdminUsername }}"
 {{- end }}
-{{- if .Values.firstRunAdminPassword }}
+{{- if or .Values.firstRunAdminPassword .Values.firstRunAdminPasswordSecret.name }}
             - name: "SEQ_FIRSTRUN_ADMINPASSWORD"
+{{- if or .Values.firstRunAdminPassword
               value: "{{ .Values.firstRunAdminPassword }}"
+{{- else }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.firstRunAdminPasswordSecret.name | quote }}
+{{- if .Values.firstRunAdminPasswordSecret.key }}
+                  key: {{ .Values.firstRunAdminPasswordSecret.key | quote }}
+{{- else }}
+                  key: password
```

Comment:
We probably shouldn't default this, and instead require it. I believe you can do this using a special [`fail`](https://masterminds.github.io/sprig/flow_control.html) function in Helm. We've got an example of that [here](https://github.com/datalust/helm.datalust.co/blob/main/charts/seq/templates/deployment.yaml#L184)